flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -63,11 +70,19 @@
 
 	private boolean isShutdown;
 
-	public NetworkEnvironment(NetworkEnvironmentConfiguration config, TaskEventPublisher taskEventPublisher) {
+	public NetworkEnvironment(
+			NetworkEnvironmentConfiguration config,
+			TaskEventPublisher taskEventPublisher,
+			TaskManagerMetricGroup taskManagerMetricGroup) {
```

Comment:
btw, do we need `TaskManagerMetricGroup`? 
Could it be `ComponentMetricGroup parentMetricGroup` to have less dependency?

Reply:
Sure, we actually use the methods of `ComponentMetricGroup`  here.